initfiles/bin/init_thor.in

```diff
@@ -143,8 +147,14 @@ while [[ 1 ]]; do
         fi
     fi
 
-    log "thormaster cmd : $instancedir/thormaster_$THORNAME MASTER=$THORMASTER:$THORMASTERPORT"
-    nohup $instancedir/thormaster_$THORNAME MASTER=$THORMASTER:$THORMASTERPORT 2> /dev/null 1>/dev/null &
+    if [ -z ${valgrindOptions} ]; then
+        log "thormaster cmd : $instancedir/thormaster_$THORNAME MASTER=$THORMASTER:$THORMASTERPORT"
+        nohup $instancedir/thormaster_$THORNAME MASTER=$THORMASTER:$THORMASTERPORT 2> /dev/null 1>/dev/null &
+    else
+        cmd="valgrind ${valgrindOptions} --log-file=$LOG_DIR/$THORNAME/valgrind.thormaster.log $instancedir/thormaster_$THORNAME MASTER=$THORMASTER:$THORMASTERPORT 2>/dev/null 1>/dev/null &"
+        log "${cmd}"
+        eval ${cmd}
+    fi
 
```

Comment:
I would probably use a pre_cmd instead of duplicating the long cmd-line. Something like:
```
pre_cmd=""
if [ -n "${valgrindOptions}" ]; then
  pre_cmd="valgrind ${valgrindOptions}"
fi
cmd=${pre_cmd} ${cmd}
...
eval ${cmd}
```